Jenkinsfile_parameterized

```diff
@@ -5,12 +5,52 @@ properties([
   parameters([
     string(name: 'PRODUCT_NAME', defaultValue: 'snl', description: ''),
     string(name: 'APP', defaultValue: 'rules', description: ''),
-    string(name: 'TYPE', defaultValue: 'java', description: ''),
-    string(name: 'ENVIRONMENT', defaultValue: 'saat', description: 'Environment where code should be build and deployed'),
-    choice(name: 'SUBSCRIPTION', choices: 'sandbox\nnonprod\nprod', description: 'Azure subscriptions available to build in')
+    string(name: 'ENVIRONMENT', defaultValue: 'sandbox', description: 'Environment where code should be built and deployed'),
+    choice(name: 'SUBSCRIPTION', choices: 'sandbox\nnonprod\nprod', description: 'Azure subscriptions available to build in'),
+    choice(name: 'VM_SIZE', choices: 'Standard_E2s_v3\nStandard_E4s_v3', description: 'Size of the VM to be created e.g. Standard_E2s_v3'),
+    string(name: 'RULES_ENGINE_SUBNET_NAME', defaultValue: 'snl-rules-engine', description: ''),
+    booleanParam(name: 'PLAN_ONLY', defaultValue: false, description: 'set to true for skipping terraform apply')
   ])
 ])
 
-withParameterizedPipeline(params.TYPE, params.PRODUCT_NAME, params.APP, params.ENVIRONMENT, params.SUBSCRIPTION) {
+product = params.PRODUCT_NAME
+component = params.APP
+environment = params.ENVIRONMENT
+subscription = params.SUBSCRIPTION
+vm_size = params.VM_SIZE
+rules_engine_subnet = params.RULES_ENGINE_SUBNET_NAME
+planOnly = params.PLAN_ONLY
+vaultName = "infra-vault-nonprod"
```

Comment:
where is that vault coming from?

Reply:
So this file was pretty much a copy of what Dan had done on the original iaas branch. That vault exists, but I haven't drilled down into the purpose or what it's there for - I had assumed it was necessary for the VM config. I'll look into the details and get back to you, same for other changes in this file.